### ISSUE TITLE
YOLO-NAS add classification head

### ISSUE THREAD

**Issue by Allison2324** [Deci-AI/super-gradients#1383]
I want to add the image classification part to YOLO-NAS model. I want to get both classification and detection predictions from the frame at once. It will be a multi-task model in which we're receiving output from encoder and having detection and classification heads further.

One of two ways I need:
1) Need to have the ability to receive output from YOLO-NAS encoder.
2) Write custom code for the new multi-task model, train loop, inference, etc.

So, does anyone have any ideas on how to solve this?

**Comment by BloodAxe** [comment]
I think what you want is to subclass from `YoloNAS` class first. Say `class YoloNasWithClassifier(YoloNAS)`.

Next step would be to add a classification head at the `__init__` method:
```
class YoloNasWithClassifier(YoloNAS):
   def __init__(self, ...):
        super().__init__(...)
        ...
        self.classification_head = ... # Depending how you want to instantiate it
```

And lastly you will need to modify forward method to also call the classification head:

```
class YoloNasWithClassifier(YoloNAS):
    def forward(self, x):
        x = self.backbone(x)
        classification_logits = self.classification_head(x)

        x = self.neck(x)
        detection_results = self.heads(x)
        return detection_results, classification_logits  
```

This is where the shallow waters end and begins a real journey. The problem is we currently do no support multi-task training and thus you would have to implement your own losses to support the new output format of this new model. Same is true for computing metrics - existing mAP and object detection metrics are not designed for multi-task objective and you will have to write a special adaptors to extract detection-specific and classification specific outputs to compute relevant metrics.

To recap:

* You would need to implement a new loss to train detection & classification head simultaneously. It can be done via classes composition.
```
class MyLoss(nn.Module):
  def __init__(self):
    self.detection_loss = PPYoloELoss(...)
    self.classification_loss = nn.CrossEntropyLoss()

  def forward(self, preds, targets)
     det_loss = self.detection_loss(preds[0], targets[0])
     cls_loss = self.classification_loss (preds[1], targets[1])
     ...
```

* Implement adapters to `DetectionMetrics` and `Accruracy` to take corresponding `[0]` or `[1]` indexes from preds and targets to compute respective metrics

* Update dataset class to return not only detection targets but also classification targets and collate function to respect that. 

That's all from top of my head so far. I may be missing something, but that should be enough to start I suppose.

**Comment by Louis-Dupont** [comment]
@Allison2324 I'm closing this issue due to inactivity. Feel free to re-open it with extra information if you are still facing issues
